Tasks/AzureFileCopyV2/PreJobExecutionAzureFileCopy.ts

```diff
@@ -0,0 +1,54 @@
+import tl = require('vsts-task-lib/task');
+import path = require('path');
+import fs = require('fs')
+import armStorage = require('azure-arm-rest/azure-arm-storage');
+import msRestAzure = require('azure-arm-rest/azure-arm-common');
+import { AzureRMEndpoint } from 'azure-arm-rest/azure-arm-endpoint';
+import { AzureEndpoint, StorageAccount } from 'azure-arm-rest/azureModels';
+
+function isNonEmpty(str: string): boolean {
+    return (!!str && !!str.trim());
+}
+
+function getResourceGroupNameFromUri(resourceUri: string): string {
+    if (isNonEmpty(resourceUri)) {
+        resourceUri = resourceUri.toLowerCase();
+        return resourceUri.substring(resourceUri.indexOf("resourcegroups/") + "resourcegroups/".length, resourceUri.indexOf("/providers"));
```

Comment:
if you don't find a match for /prodivers, your substring construction is likely misbehave. Can you add UT around this with boundary conditions and make this string extraction more defensive.

Reply:
Any reason you did not use match() with regex to extract resourcename between known strings "resourcegroups/" and "/providers"?